src/content/en/updates/2019/08/barcode-detection.md

```diff
@@ -0,0 +1,128 @@
+
+### Working With the `BarcodeDetector` {: #barcodedetector}
+
+Let's briefly recall how working with the `BarcodeDetector` looks like:
+
+```js
+const barcodeDetector = new BarcodeDetector();
+try {
+  const barcodes = await barcodeDetector.detect(image);
+  barcodes.forEach(barcode => console.log(barcode));
+} catch (err) {
+  console.error('Barcode detection failed:', err);
+}
+```
+
+## Current Status {: #status }
+
+| Step                                       | Status                               |
+| ------------------------------------------ | ------------------------------------ |
+| 1. Create explainer                        | [Complete][explainer]                |
+| 2. Create initial draft of specification   | [Complete][spec]                     |
+| 3. Gather feedback & iterate on design     | [Complete][wicg-discourse]           |
+| 4. Origin trial                            | [Complete][ot]                       |
+| 5. **Launch**                              | **Complete**                         |
+
+### Origin Trial Feedback Summary
+
+The primary feedback during the Origin Trial was around the limited availability
+of barcode detection support across the various Blink platforms.
+In response to this feedback, we've now added a new static `getSupportedFormats()` method
+to more easily allow developers to detect when the current platform supports
+the desired formats, and when a polyfill is necessary.
+We've also hooked up the `formats` hint options parameter to the underlying implementation,
+which allows it to optimize for detecting only the formats specified.
+The code sample below shows how the new method could be used to detect QR codes
+(among several
+[other possible barcode formats](https://wicg.github.io/shape-detection-api/#enumdef-barcodeformat)).
+
+```js
+// 🆕 Check if the platform supports QR codes.
+if ((await BarcodeDetector.getSupportedFormats()).includes('qr_code')) {
+  // Initialize the barcode detector. As this could be costly,
+  // do this only once and from thereon reuse the instance.
+  const barcodeDetector = new BarcodeDetector({
+    // Hint at the format(s) we're after, allowing the implementation
+    // to optimize and only run detection for the mentioned format(s).
+    formats: ['qr_code']
+  });
+  try {
+    // Try to decode QR codes.
+    const detectedCodes = await barcodeDetector.detect(someImage);
+    console.log(detectedCodes);
+  } catch (err) {
+    // Handle potential errors.
+    console.error(err);
+  }
+} else {
+  // Load a polyfill in form of a JavaScript-based QR code library.
+}
+```
+
+### Barcode Detection Demo
+
+To see the new `getSupportedFormats` method (and of course barcode detection itself) in action,
+head over to this [simple demo][demo].
+For a more advanced real-world Progressive Web App
+that you can actually install to your device's home screen,
+check out [QR Snapper](https://qrsnapper.com/).
```

Comment:
Do we also want to mention [Perception Toolkit](https://perceptiontoolkit.dev/)?

Reply:
Great suggestion, added.